janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -100,9 +100,14 @@ public String toString() {
 
     @Override
     public JointIndexQuery updateLimit(int newLimit) {
-        JointIndexQuery ji = new JointIndexQuery(Lists.newArrayList(queries));
-        ji.setLimit(newLimit);
-        return ji;
+        JointIndexQuery queryWithNewLimit;
```

Comment:
IMO looks strange. Consider if doing it just like:
```suggestion
        _if (queries.size() == 1) queries.get(0).updateLimit(newLimit);_
        JointIndexQuery ji = new JointIndexQuery(Lists.newArrayList(queries));
        ji.setLimit(newLimit);
        return ji;
```
before isn't a better option. Moreover, it may be more readable to add dedicated method to that handles this line.